apps/minifront/src/state/status.ts

```diff
@@ -1,36 +1,37 @@
-import { SliceCreator } from '.';
-import { viewClient } from '../clients';
+import { ZQueryState } from '@penumbra-zone/zquery/src/types';
+import { SliceCreator, useStore } from '.';
+import { createZQuery } from '@penumbra-zone/zquery';
+import { getStatusStream } from '../fetchers/status';
 
-export interface StatusSlice {
-  start: () => Promise<void>;
+interface Status {
   fullSyncHeight?: bigint;
   latestKnownBlockHeight?: bigint;
-  error?: unknown;
 }
 
-export const createStatusSlice = (): SliceCreator<StatusSlice> => set => ({
-  start: async () => {
-    try {
-      // statusStream sends new data to stream only when a new block is detected.
-      // This can take up to 5 seconds (time of new block generated).
-      // Therefore, we need to do a unary request to start us off.
-      const status = await viewClient.status({});
-      set(state => {
-        state.status.fullSyncHeight = status.fullSyncHeight;
-        if (!status.catchingUp) state.status.latestKnownBlockHeight = status.fullSyncHeight;
-      });
-
-      const stream = viewClient.statusStream({});
-      for await (const result of stream) {
-        set(state => {
-          state.status.fullSyncHeight = result.fullSyncHeight;
-          state.status.latestKnownBlockHeight = result.latestKnownBlockHeight;
-        });
-      }
-    } catch (error) {
-      set(state => {
-        state.status.error = error;
-      });
-    }
+export const { status, useStatus } = createZQuery({
+  name: 'status',
+  fetch: getStatusStream,
+  stream: (_prevState: Status | undefined, item: Status): Status => ({
+    fullSyncHeight: item.fullSyncHeight,
+    latestKnownBlockHeight: item.latestKnownBlockHeight,
+  }),
+  getUseStore: () => useStore,
```

Comment:
[question](https://conventionalcomments.org/): why do we actually need the whole store for manipulating with just one slice? Aren't getters and setters enough?

Reply:
Ugh, unfortunately we need it for actually calling the hook (e.g., [here](https://github.com/penumbra-zone/web/blob/jessepinho/z-query-web-965/packages/zquery/src/index.ts#L185-L195)). I wanted to avoid this but, since we're actually hooking into the React component lifecycle, we need to actually use the `useStore` hook (rather than just the getter/setter).